### DISCUSSION TITLE
Using 2 results sources as one in the same search results webpart

### DISCUSSION THREAD

**Discussion by Jrstew2016** [microsoft-search/pnp-modern-search#3312] [Q&A]
Is there a way to use two result sources such as People results and SP results? Or if not. Can a custom result source be created and then queried somehow I'm the search results box?
And if either of those cannot be done. Can creating a custom data connection achieve this?
We are building out a custom people directory that will be used for several groups with different fields along with pulling in user profile properties such as title, picture, and a couple other attributes along with several custom fields. I don't want to use the SP user Profile properties because each directory is going to have different fields and I'd prefer not to have 100s of extra properties added there. So I am using custom lists for the info to be pulled from. Then use a custom people card which we've already developed.

**Comment by kasperbolarsen** [comment]
Using SharePoint search I doubt you can use both lndexes. However, Microsoft Search allows combinations of entity types, so that might be an option,  https://learn.microsoft.com/en-us/graph/search-concept-multiplerequest

**Reply by Jrstew2016** [reply]
Unfortunately we don't have anyone with Graph experience and they're reluctant to give us access even to work with Graph. And you say even a custom connection wouldn't work?

**Reply by kasperbolarsen** [reply]
I am NOT saying that SharePoint search is about to be deprecated.....at least not officially,  but the writing on the wall is pretty clear. 
This is also why PnP Modern Search web parts now supports both SharePoint Search and Microsoft Search as data sources.
My advice would be to skill up on Graph/ Microsoft search,  and one way to convince management to provide the resources could be to demo Graph Connectors as these provide additional data to both search AND AI/Copilot 😀 
Google or check the app formerly known as Twitter for content by @waldekm 

I haven't had to resolve to custom providers yet, but I haven't been able to find any sample doing that, so I doubt.
